src/libguac/protocol.c

```diff
@@ -961,6 +961,37 @@ int guac_protocol_send_rect(guac_socket* socket,
 
 }
 
+static int __guac_protocol_send_required(guac_socket* socket,
```

Comment:
Please document this function.

Also if sending a NULL-terminated array of strings as a series of instruction elements is common here (it looks like we've done this with at least `args` and `connect`, as well), I think it's worth considering whether we should extract the commonality so there is one internal function leveraged by all three, similar to how we have `__guac_socket_write_length_string()` and `__guac_socket_write_length_int()`.

Reply:
Documentation added.  Also implemented common function `__guac_socket_write_array()` for writing out a NULL-terminated array.